src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -409,8 +409,32 @@ class LiteralCoercionSegment(BaseSegment):
     )
 
 
+# Dialects should not use Python "import" to access other dialects. Instead,
+# get a reference to the ANSI ObjectReferenceSegment this way so we can inherit
+# from it.
+ObjectReferenceSegment = ansi_dialect.get_segment("ObjectReferenceSegment")
+
+
+@bigquery_dialect.segment(replace=True)
+class ColumnReferenceSegment(ObjectReferenceSegment):  # type: ignore
+    """A reference to column, field or alias."""
+
+    type = "column_reference"
+
+    def extract_possible_references(self, level: int):
+        """Extract possible references of a given level."""
+        refs = list(self.iter_raw_references())
+        if level == 3 and len(refs) >= 3:
+            return [refs[0]]
+        if level == 2 and len(refs) >= 3:
```

Comment:
can you replace the values with the enum now?

Reply:
Done! I also made some other changes, hinted at in my comment above.